### DISCUSSION TITLE
two python scripts monitoring the same gpio pin at the same time

### DISCUSSION THREAD

**Discussion by gldrplt** [gpiozero/gpiozero#1152] [Q&A]
I have two Python programs running simultaneously that watch the same gpio pin.  These programs work on a Raspberry Pi Zero running Python version 3.9.2.  Both programs use gpiozero. No virtual environments involved
However, I just installed a Raspberry Pi 5 running Bookworm.  I set up a venv and installed gpiozero.  Each Python program runs individually, but when I run them together the second program gets a GPIO Busy error.

Is there a way  to do this?
Thanks

**Comment by lurch** [comment]
In Bookworm the GPIO pins are controlled by the Linux kernel and libgpiod, and IIRC the kernel only allows one program at a time to talk to each GPIO pin.
So I think you'll have to have only one Python script watching the GPIO pin, and use some kind of Inter-Process-Communication (IPC) to talk to your other Python script.

**Reply by TheExpertOne** [reply]
Why did Raspberry Pi did this? Sorry to say that, this solution is not going to work for me.

**Comment by gldrplt** [comment]
Thanks for the prompt reply.
